src/peft/utils/save_and_load.py

```diff
@@ -100,6 +118,23 @@ def get_peft_model_state_dict(model, state_dict=None, adapter_name="default", un
             if any(f"{module_name}.modules_to_save.{adapter_name}" in key for module_name in model.modules_to_save):
                 to_return[key.replace("modules_to_save.", "")] = value
 
+    # check the common embedding layers in `target_modules` to reset `is_embedding_layer_resized` if necessary
+    if (
+        hasattr(config, "target_modules")
+        and any(k in config.target_modules for k in EMBEDDING_LAYER_NAMES)
+        and not is_embedding_layer_resized
+    ):
+        warnings.warn("Setting `is_embedding_layer_resized` to `True` as embedding layers found in `target_modules`")
```

Comment:
> Additionally, for users who are using prompt-learning, this is not automatically detected, right? Would it make sense to do that too?

Couldn't find a way to do it for prompt learning methods as there is no config related to embeddings being targeted or resized.

Reply:
> Of course, then it would not work automatically anymore, which was the initial goal.

Yes, and then users can't simply use `trainer.push_to_hub()` or `trainer.save_model()` which adds more burden on end-user. I think the current implementation is fine even though we might save embedding when they weren't resized but targeted. Most usecases will add loras to embedding layers when they resize it.

Reply:
So what about having the options `True` and `"auto"` (or something similar) which work like `True` and `False` now, with `"auto"` being the default, and if a user explicitly passes `False`, we won't save the embeddings.

Reply:
Done, added `auto`